src/subgame.cpp

```diff
@@ -223,15 +221,31 @@ std::string getWorldGameId(const std::string &world_path, bool can_be_legacy)
 	return conf.get("gameid");
 }
 
+std::string getWorldPathEnv()
+{
+	char *world_path = getenv("MINETEST_WORLD_PATH");
+
+	if (world_path)
+		return std::string(world_path);
+	return "";
+}
+
 std::vector<WorldSpec> getAvailableWorlds()
 {
 	std::vector<WorldSpec> worlds;
 	std::set<std::string> worldspaths;
+
+	Strfnd search_paths(getWorldPathEnv());
+
+	while(!search_paths.atend()) {
+		worldspaths.insert(search_paths.next(PATH_DELIM));
+	}
```

Comment:
I noticed above you did not enclose single-line if statements in brackets, but you do for this while statement.  Either is fine, but you should stay consistent within your own code style.